.github/workflows/changelog.yml

```diff
@@ -0,0 +1,51 @@
+name: Changelog
+
+on:
+  push:
+    branches:
+      # todo: change this branch name
+      - master
+    repository_dispatch:
+      types: [semantic-release]
```

Comment:
Can you talk about what this does? I'm not sure I fully understand how `repository_dispatch` works.

Reply:
I brought this over from what the OSS was doing. I believe it's required to allow actions to have the right scope for writing to the repository, but I'm not clear. [Here](https://github.com/marketplace/actions/repository-dispatch) is the action details.